lib/improver/blending/weighted_blend.py

```diff
@@ -113,6 +113,8 @@ def conform_metadata(
                 forecast_period_coord(cube,
                                       force_lead_time_calculation=True))
             forecast_period.bounds = None
+            forecast_period.convert_units(cube.coord("forecast_period").units)
```

Comment:
Pretty sure forecast period should always be in seconds, which is what the forecast_period_coord plugin above returns. Does it need to match forecast_period for some reason?

Reply:
This preserves the behaviour in our tests - some of those have forecast_period in hours.